src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -35,155 +46,608 @@ def deserialize_bom_json(file_path: Path) -> dict:
 
     Returns
     -------
-    dict
-        The BOM content as a dictionary.
+    Bom
+        The CycloneDX Bom object.
 
     Raises
     ------
     CycloneDXParserError
         If the bom.json file cannot be located or deserialized.
-
     """
-    # TODO: use the official CycloneDX library to validate and deserialize BOM files
-    # once merged: https://github.com/CycloneDX/cyclonedx-python-lib/pull/290
     if not os.path.exists(file_path):
         raise CycloneDXParserError(f"Unable to locate any BOM files at: {str(file_path.parent)}.")
 
+    # We use the `cyclonedx-python-library` library for deserialization following the example here:
+    # https://cyclonedx-python-library.readthedocs.io/en/v7.3.4/examples.html
     with open(file_path, encoding="utf8") as file:
+        json_data = file.read()
+        if defaults.getboolean("dependency.resolver", "validate", fallback=True):
+            try:
+                schema_version = defaults.get("dependency.resolver", "schema", fallback="1.6")
+                my_json_validator = JsonStrictValidator(SchemaVersion.from_version(schema_version))
+            except ValueError as error:
+                raise CycloneDXParserError(f"Unable to find schema validator for {schema_version}: {error}") from error
+            try:
+                validation_errors = my_json_validator.validate_str(json_data)
+            except MissingOptionalDependencyException as error:
+                # Allow MissingOptionalDependencyException.
+                logger.debug("JSON-validation was skipped due to %s", error)
+            except json.JSONDecodeError as error:
+                raise CycloneDXParserError(f"BOM file is invalid: {error}") from error
```

Comment:
Looking at the documentation for `validate_str` of the JsonStrictValidator https://cyclonedx-python-library.readthedocs.io/en/latest/autoapi/cyclonedx/validation/json/index.html#cyclonedx.validation.json.JsonStrictValidator
I have the following understanding:
- If `json_data` is not a valid JSON content -> `json.JSONDecodeError` is raised from `validate_str`
- If `json_data` is a valid JSON content, but violates the schema for cyclonedx, those errors will be returned as `validation_errors`
Is my assumption correct?